src/actions/PrecursorAction.C

```diff
@@ -269,5 +311,60 @@ PrecursorAction::act()
         _problem->addInitialCondition("RandomIC", ic_name, params);
       }
     }
+
+    if (_current_task == "add_postprocessor" && getParam<bool>("loop_precs"))
+    {
+        // looping precursors requires connecting outlet of core problem
+        // to the inlet of the loop subproblem. In addition, the outlet of the
+        // loop must be connected to the core problem.
+        {
+            std::string postproc_name = "Outlet_SideAverageValue_"+var_name+"_"+_object_suffix;
+            InputParameters params = _factory.getValidParams("SideAverageValue");
+            std::vector<VariableName> varvec(1);
+            varvec[0] = var_name;
+            params.set<std::vector<VariableName>>("variable") = varvec;
+            params.set<std::vector<BoundaryName>>("boundary") =
+                getParam<std::vector<BoundaryName>>("outlet_boundaries");
+            
+            _problem->addPostprocessor("SideAverageValue", postproc_name, params);
+        }
+        {
+            std::string postproc_name = "Inlet_SideAverageValue_"+var_name+"_"+_object_suffix;
+            InputParameters params = _factory.getValidParams("Receiver");
+            params.set<MultiMooseEnum>("execute_on")= "timestep_begin";
+            
+            _problem->addPostprocessor("Receiver", postproc_name, params);
+        }
+        
+    }
+
+    if (_current_task == "add_transfer" && getParam<bool>("loop_precs") &&
+            !getParam<bool>("is_loopapp"))
+    {
+        // from main app to loop app
```

Comment:
are you just copying variables over? how often does this happen?

Reply:
Once, in problem setup. These set up Transfer objects that carry reactor core outlet precursor concentrations to the loop inlet at the start of each timestep. Same goes for loop outlet into core inlet.